// @siberiacancode/reactuse/dist/esm/hooks/useGeolocation/useGeolocation.mjs
import { useState as o, useEffect as r } from "react";
const d = (e) => {
  const [a, n] = o({
    loading: !0,
    error: null,
    timestamp: Date.now(),
    accuracy: 0,
    latitude: Number.POSITIVE_INFINITY,
    longitude: Number.POSITIVE_INFINITY,
    altitude: null,
    altitudeAccuracy: null,
    heading: null,
    speed: null
  });
  return r(() => {
    const u = ({ coords: t, timestamp: c }) => {
      n({
        ...a,
        loading: !1,
        timestamp: c,
        latitude: t.latitude,
        longitude: t.longitude,
        altitude: t.altitude,
        accuracy: t.accuracy,
        altitudeAccuracy: t.altitudeAccuracy,
        heading: t.heading,
        speed: t.speed
      });
    }, l = (t) => {
      n({
        ...a,
        loading: !1,
        error: t
      });
    };
    navigator.geolocation.getCurrentPosition(u, l, e);
    const i = navigator.geolocation.watchPosition(u, l, e);
    return () => {
      navigator.geolocation.clearWatch(i);
    };
  }, [e?.enableHighAccuracy, e?.maximumAge, e?.timeout]), a;
};
export {
  d as useGeolocation
};
