// @siberiacancode/reactuse/dist/esm/hooks/useBluetooth/useBluetooth.mjs
import { useState as n, useEffect as D } from "react";
const h = (d) => {
  const o = typeof navigator < "u" && "bluetooth" in navigator, { acceptAllDevices: l = !1, filters: c, optionalServices: u } = d ?? {}, [f, s] = n(!1), [e, r] = n(void 0), [g, i] = n(void 0), p = async () => {
    if (!o) return;
    const t = await navigator.bluetooth.requestDevice({
      acceptAllDevices: l,
      optionalServices: u,
      ...c && { filters: c, acceptAllDevices: !1 }
    });
    r(t);
  };
  return D(() => {
    if (e && e.gatt) {
      const t = async () => {
        if (!e.gatt) return;
        const v = await e.gatt.connect();
        i(v), s(v.connected);
      }, a = () => {
        i(void 0), r(void 0), s(!1);
      };
      return e.addEventListener("gattserverdisconnected", a), t(), () => {
        e.removeEventListener("gattserverdisconnected", a), e.gatt?.disconnect();
      };
    }
  }, [e]), {
    supported: o,
    connected: f,
    device: e,
    requestDevice: p,
    server: g
  };
};
export {
  h as useBluetooth
};
