// @siberiacancode/reactuse/dist/esm/hooks/useBreakpoints/useBreakpoints.mjs
import { useEffect as u } from "react";
import { useRerender as I } from "../useRerender/useRerender.mjs";
const l = (s) => window.matchMedia(s).matches, N = (s, d = "mobile-first") => {
  const a = I(), r = (e, t) => t ? `${Number(s[e]) + t}px` : `${s[e]}px`;
  u(() => {
    const e = () => a();
    return window.addEventListener("resize", e), () => {
      window.removeEventListener("resize", e);
    };
  }, []);
  const n = (e) => l(`(min-width: ${r(e)})`), m = (e) => l(`(max-width: ${r(e)})`), o = () => Object.keys(s).map((e) => [e, n(e)]).filter(([, e]) => e).map(([e]) => e), E = () => {
    const e = o();
    return e.length ? e.at(-1) : void 0;
  }, i = (e) => l(`(min-width: ${r(e, 0.1)})`), c = (e) => l(`(max-width: ${r(e, -0.1)})`), x = (e, t) => l(`(min-width: ${r(e)}) and (max-width: ${r(t, -0.1)})`), A = Object.keys(s).reduce(
    (e, t) => {
      const S = d === "mobile-first" ? n(t) : m(t);
      return e[t] = S, e;
    },
    {}
  );
  return {
    current: o,
    greaterOrEqual: n,
    smallerOrEqual: m,
    active: E,
    greater: i,
    smaller: c,
    between: x,
    isGreater: i,
    isGreaterOrEqual: n,
    isSmaller: c,
    isSmallerOrEqual: m,
    isInBetween: x,
    ...A
  };
}, R = {
  xs: 0,
  sm: 600,
  md: 900,
  lg: 1200,
  xl: 1536
}, _ = {
  xs: 576,
  sm: 768,
  md: 992,
  lg: 1200,
  xl: 1408
}, w = {
  sm: 640,
  md: 768,
  lg: 1024,
  xl: 1280,
  "2xl": 1536
}, g = {
  xs: 0,
  sm: 576,
  md: 768,
  lg: 992,
  xl: 1200,
  xxl: 1400
}, B = {
  xs: 480,
  sm: 576,
  md: 768,
  lg: 992,
  xl: 1200,
  xxl: 1600
}, K = {
  xs: 0,
  sm: 600,
  md: 1024,
  lg: 1440,
  xl: 1920
}, P = {
  mobileS: 320,
  mobileM: 375,
  mobileL: 425,
  tablet: 768,
  laptop: 1024,
  laptopL: 1440,
  desktop4K: 2560
}, h = {
  "3xs": 360,
  "2xs": 480,
  xs: 600,
  sm: 768,
  md: 1024,
  lg: 1280,
  xl: 1440,
  "2xl": 1600,
  "3xl": 1920,
  "4xl": 2560
}, f = {
  sm: 576,
  md: 768,
  lg: 992,
  xl: 1200
};
export {
  B as BREAKPOINTS_ANT_DESIGN,
  g as BREAKPOINTS_BOOTSTRAP_V5,
  _ as BREAKPOINTS_MANTINE,
  h as BREAKPOINTS_MASTER_CSS,
  R as BREAKPOINTS_MATERIAL_UI,
  f as BREAKPOINTS_PRIME_FLEX,
  K as BREAKPOINTS_QUASAR_V2,
  P as BREAKPOINTS_SEMANTIC,
  w as BREAKPOINTS_TAILWIND,
  N as useBreakpoints
};
